### PR TITLE
Master wowl searchpanel

### PR DESCRIPTION
**Pull request by Arcasias**
Enterprise: https://github.com/odoo-dev/enterprise/pull/210

### CODE REVIEW

File: addons/web/static/src/search/search_model.js [hunk 1]
```diff
@@ -1403,6 +1385,25 @@ export class SearchModel extends EventBus {
         return dateFilterRange[key];
     }
 
+    /**
+     * Returns which components are displayed in the current view. Components
```
Comment (Polymorphe57) [inline addons/web/static/src/search/search_model.js]: possibly no current view (case client action using a WithSearch)

File: addons/web/static/src/search/search_panel/search_panel.js [hunk 2]
```diff
@@ -72,22 +72,21 @@ export class SearchPanel extends Component {
     }
 
     //---------------------------------------------------------------------
-    // Private
+    // Protected
```
Comment (Polymorphe57) [inline addons/web/static/src/search/search_panel/search_panel.js:75]: not in component I think

File: addons/crm/views/crm_lead_views.xml [hunk 3]
```diff
@@ -700,6 +700,9 @@
                         <filter string="Creation Date" context="{'group_by':'create_date:month'}" name="month"/>
                         <filter string="Closed Date" name="date_closed" context="{'group_by':'date_closed'}"/>
                     </group>
+                    <searchpanel view_types="kanban,graph">
```
Comment (Polymorphe57) [inline addons/crm/views/crm_lead_views.xml]: don't forget to remove

File: addons/web/static/src/search/search_arch_parser.js [hunk 4]
```diff
@@ -160,7 +172,9 @@ export class SearchArchParser extends XMLParser {
                         preSearchItem.defaultIntervalId = defaultInterval || DEFAULT_INTERVAL;
                     }
                 }
-            } catch (e) {}
+            } catch (e) {
```
Comment (aab-odoo) [inline addons/web/static/src/search/search_arch_parser.js]: what could crash here? We should only put in the try {} what can throw the error we want to catch

File: addons/web/static/src/search/with_search/with_search.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
 /** @odoo-module **/
 
+import { useBus, useEffect, useService } from "@web/core/utils/hooks";
+// import { ControlPanel } from "@web/search/control_panel/control_panel";
```
Comment (aab-odoo) [inline addons/web/static/src/search/with_search/with_search.js]: remove

File: addons/web/static/src/search/search_arch_parser.js [hunk 6]
```diff
@@ -148,20 +175,17 @@ export class SearchArchParser extends XMLParser {
...
-                if (groupBy) {
-                    preSearchItem.type = "groupBy";
-                    const [fieldName, defaultInterval] = makeContext(context).group_by.split(":");
-                    preSearchItem.fieldName = fieldName;
-                    preSearchItem.fieldType = this.fields[fieldName].type;
-                    if (["date", "datetime"].includes(preSearchItem.fieldType)) {
-                        preSearchItem.type = "dateGroupBy";
-                        preSearchItem.defaultIntervalId = defaultInterval || DEFAULT_INTERVAL;
-                    }
+            const [fieldName, defaultInterval] = getContextGroubBy(context);
```
Comment (Polymorphe57) [inline addons/web/static/src/search/search_arch_parser.js:178]: nope: does not work on "[foo,bar]" I think

**Pull request closed without merging**